src/Dotnet.Script.DependencyModel/Context/ScriptDependencyInfoProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Dotnet.Script.DependencyModel.Logging;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Dotnet.Script.DependencyModel.Context
+{
+    /// <summary>
+    /// Represents a class that is capable of providing 
+    /// the <see cref="DependencyContext"/> for a given project file (csproj).
+    /// </summary>
+    public class ScriptDependencyInfoProvider 
+    {
+        private readonly IRestorer[] _restorers;
+        private readonly Logger _logger;
+
+        public ScriptDependencyInfoProvider(IRestorer[] restorers, LogFactory logFactory)
+        {
+            _restorers = restorers;
+            _logger = logFactory.CreateLogger<ScriptDependencyInfoProvider>();
+        }
+
+        /// <summary>
+        /// Gets the <see cref="DependencyContext"/> for the project file.
+        /// </summary>
+        /// <param name="pathToProjectFile">The path to the project file.</param>
+        /// <returns>The <see cref="DependencyContext"/> for a given project file (csproj).</returns>
+        public ScriptDependencyInfo GetDependencyInfo(string pathToProjectFile)
+        {            
+            Restore(pathToProjectFile);
+            var context =  ReadDependencyContext(pathToProjectFile);
+            var nugetPackageFolders = GetNuGetPackageFolders(pathToProjectFile);
+            return new ScriptDependencyInfo(context, nugetPackageFolders);
+        }
+
+        private void Restore(string pathToProjectFile)
+        {
+            foreach (var restorer in _restorers)
+            {
+                if (restorer.CanRestore)
```

Comment:
it looks like in most cases, this will run both restorers, is this intended?

Reply:
oh sorry, I meant it rather will only run the first restorer

Reply:
That is intentional. The 99% case here would be the `DotnetRestorer`. The `NuGetRestorer` is sort of a fallback in the case where a script containing NuGet references is opened in OmniSharp without the `dotnet sdk` installed. In real life this would be a rare scenario